$MirOS: ports/www/lynx/patches/patch-samples_oldlynx,v 1.2 2007/05/20 12:06:19 tg Exp $

	rewritten something more sane and secure
	but requires mksh for execution

--- samples/oldlynx.orig	Tue Sep 19 00:28:05 2006
+++ samples/oldlynx	Sun Mar  9 17:50:39 2008
@@ -1,20 +1,21 @@
-#!/bin/sh
-# invoke lynx built with color-style, overriding the color options to use the
-# non-color-style scheme -TD
+#!/usr/bin/env mksh
+# $MirOS: ports/www/lynx/patches/patch-samples_oldlynx,v 1.2 2007/05/20 12:06:19 tg Exp $
 
-my_cfg=${TMPDIR:-/tmp}/lynxcfg$$
-trap "rm -f $my_cfg" 0 1 2 5 15
+T=$(mktemp ${TMPDIR:-/tmp}/lynx.cfg.XXXXXXXXXX) || {
+	print -u2 Fatal: cannot create temporary file!
+	exit 1
+}
+trap 'rm -f $T ; exit 0' 0
+trap 'rm -f $T ; exit 1' 1 2 3 5 13 15
 
-rm -f "$my_cfg"
-echo "DEFAULT_COLORS:off" >>$my_cfg
-if test -n "$LYNX_CFG" ; then
-	echo "include:$LYNX_CFG" >>$my_cfg
-fi
-echo "COLOR_STYLE:" >>$my_cfg
-echo "NESTED_TABLES:off" >>$my_cfg
+print DEFAULT_COLORS:off >$T
+[[ -n $LYNX_CFG && -e $LYNX_CFG ]] && print -r "include:$LYNX_CFG" >>$T
+print COLOR_STYLE: >>$T
+print NESTED_TABLES:off >>$T
 
-LYNX_CFG=$my_cfg
-export LYNX_CFG
+export LYNX_CFG=$T
 unset LYNX_LSS
-
 ${LYNX_PROG-lynx} "$@"
+rv=$?
+rm -f $T
+exit $rv
